orderfilter/validate.go

```diff
@@ -0,0 +1,55 @@
+// +build !js
+
+package orderfilter
+
+import (
+	"context"
+
+	"github.com/0xProject/0x-mesh/zeroex"
+	peer "github.com/libp2p/go-libp2p-core/peer"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	log "github.com/sirupsen/logrus"
+	jsonschema "github.com/xeipuuv/gojsonschema"
+)
+
+func (f *Filter) ValidateOrderJSON(orderJSON []byte) (*jsonschema.Result, error) {
+	return f.orderSchema.Validate(jsonschema.NewBytesLoader(orderJSON))
+}
+
+// MatchOrder returns true if the order passes the filter. It only returns an
+// error if there was a problem with validation. For details about
+// orders that do not pass the filter, use ValidateOrder.
+func (f *Filter) MatchOrder(order *zeroex.SignedOrder) (bool, error) {
```

Comment:
I think `MatchOrder` can be shared between Go and JS/Wasm? It just calls `ValidateOrder` and `reult.Valid`.